app/utils/injectSaga.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import getInjectors from './sagaInjectors';
+
+/**
+ * Dynamically injects a saga, passes component's props as saga arguments
+ *
+ * @param {string} name Name of the saga
+ * @param {function} saga A root saga that will be injected
+ * @param {string} [mode] By default ('restart-on-remount') the saga will be started on component mount and cancelled with
```

Comment:
I've created one set of constants for saga and reducer injectors for simplicity but don't know if it's simple or simplistic, I can split them in separate files if necessary.